alchemist/alchemist-fxui/src/main/java/it/unibo/alchemist/boundary/gui/view/SingleRunApp.java

```diff
@@ -0,0 +1,417 @@
+ * @param <P> the position type
+ */
+@SuppressFBWarnings(
+        value = "DM_EXIT",
+        justification = "The program needs to exit when the user clicks on the exit button"
+)
+public class SingleRunApp<T, P extends Position2D<P>> extends Application {
+    /**
+     * Main layout without nested layouts. Must inject eventual other nested layouts.
+     */
+    public static final String ROOT_LAYOUT = "RootLayout";
+    /**
+     * Effect pass param name.
+     */
+    public static final String USE_EFFECT_GROUPS_FROM_FILE = "use-effect-groups-from-file";
+    /**
+     * Default parameter start string.
+     */
+    public static final String PARAMETER_NAME_START = "--";
+    /**
+     * Default parameter end string.
+     */
+    public static final String PARAMETER_NAME_END = "=";
+    /**
+     * Default logger for the class.
+     */
+    private static final Logger L = LoggerFactory.getLogger(SingleRunApp.class);
+
+    private final Map<String, String> params = new HashMap<>();
+    private ObservableList<EffectGroup<P>> effectGroups = FXCollections.observableArrayList();
+    private boolean initialized;
+    @Nullable
+    private Simulation<T, P> simulation;
+    @Nullable
+    private FXOutputMonitor<T, P> displayMonitor;
+    private PlayPauseMonitor<T, P> playPauseMonitor;
+    private FXTimeMonitor<T, P> timeMonitor;
+    private FXStepMonitor<T, P> stepMonitor;
+
+    /**
+     * Getter method for the named parameters.
+     *
+     * @return the named params
+     */
+    private Map<String, String> getParams() {
+        if (params.isEmpty()) {
+            Optional.ofNullable(getParameters()).ifPresent(p -> params.putAll(p.getNamed()));
+        }
+        return this.params;
+    }
+
+    /**
+     * The method adds a new named parameter.
+     *
+     * @param name  the param name
+     * @param value the param value
+     * @throws IllegalArgumentException if the parameter is not valid,
+     *      or if {@link Parameters#getUnnamed()} it's not named}
+     * @throws IllegalStateException    if the application is already started
+     * @see Parameters#getNamed()
+     */
+    public void addParam(final String name, final String value) {
+        checkIfInitialized();
+        if (value == null || value.equals("")) {
+            throw new IllegalArgumentException("The given param is not named");
+        }
+        params.put(name, value);
+    }
+
+    /**
+     * The method sets the parameters. All previously add params will be removed.
+     *
+     * @param params the params
+     * @throws IllegalStateException if the application is already started
+     * @see Application#getParameters()
+     */
+    public void setParams(final String[] params) {
+        if (initialized) {
+            throw new IllegalStateException("Application is already initialized");
+        }
+        this.params.clear();
+        Arrays.stream(params)
+                .forEach(p -> {
```

Comment:
Do not use streams for a single foreach. It's better to use a classic for, this is just (irrelevant) overhead and harder debug